spec/openhab/dsl/rules/builder_spec.rb

```diff
@@ -1069,35 +1083,90 @@ def self.generate(name, cron_expression, *every_args, attach: nil, **kwargs)
       around do |example|
         Dir.mktmpdir("openhab-rspec") do |dir|
           @temp_dir = dir
+          file_counter += 1
+          test_file = "rspec-test-#{file_counter}"
```

Comment:
- I haven't tried using "let" because I read that it memoize the result. Plus I reuse the variable in the rest to build absolute path and extension. It's no biggie if we're using let I could just use a different variable name for that though. The memoization is the deal breaker. Once again I haven't tried it.
- random was the first thing I thought of, but compared to sequential, it still carries a tiny risk of producing two consecutive numbers that are the same, since there's no contract to say they must be different on each invocation.

Reply:
`let` memoizes for each spec. it does _not_ keep the same value between specs. then you just access it (mostly) like a local variable in each spec.

with a length of 8, that's a 1 in 62^8 chance of having duplicates. effectively impossible.

Reply:
I think I'll take up your suggestion and use securerandom.

Reply:
> `let` memoizes for each spec. it does _not_ keep the same value between specs. then you just access it (mostly) like a local variable in each spec.

Ohhh that sounds perfect! I'll change it to `let`.